### PR TITLE
BREAKING CHANGE: feat(#2135) Validate pipeline template configuration

### PR DESCRIPTION
**Pull request by klu909**
## Context
this is a replicate of https://github.com/screwdriver-cd/template-validator/pull/32 , adding test and fixing some error

## Objective

add validation for pipeline template
## References

https://github.com/screwdriver-cd/screwdriver/issues/2135

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by klu909** [comment]
weren't able to add test for `locked steps` and `parent template` because we do not support these feature yet

**Comment by pritamstyz4ever** [comment]
Lets add BREAKING_CHANGE and bump major version

**Review by pritamstyz4ever — COMMENTED** [review]
Lets add BREAKING_CHANGE and bump major version

File: index.js [hunk 1]
```diff
@@ -118,4 +120,41 @@ async function parseTemplate(yamlString, templateFactory) {
     }
 }
 
-module.exports = parseTemplate;
+/**
+ * Parses the pipeline configuration from a screwdriver-template.yaml
+ * @method parseTemplate
```
Comment (tkyi) [inline index.js]:
```suggestion
 * @method parsePipelineTemplate
```

**Review by pritamstyz4ever — APPROVED** [review]

**Pull request merged**